### PR TITLE
Updates to the Dependencies Handling

### PR DESCRIPTION
**Pull request by NikCharlebois**
#### Pull Request (PR) description
Added a new mechanism to handle dependencies and removed the required modules list from the main module's manifest.

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

File: Modules/Microsoft365DSC/Modules/M365DSCLogEngine.psm1 [hunk 1]
```diff
@@ -28,7 +28,10 @@ function New-M365DSCLogEntry
     {
         Write-Host "$($Global:M365DSCEmojiRedX)"
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Modules/M365DSCLogEngine.psm1:32]: Indentation inconsistent

File: Modules/Microsoft365DSC/Modules/M365DSCReport.psm1 [hunk 2]
```diff
@@ -327,7 +330,10 @@ function Compare-M365DSCConfigurations
 
     if ($CaptureTelemetry)
     {
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Modules/M365DSCReport.psm1:334]: Indentation inconsistent

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADNamedLocationPolicy/MSFT_AADNamedLocationPolicy.psm1 [hunk 3]
```diff
@@ -388,6 +394,9 @@ function Export-TargetResource
     )
     $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' -InboundParameters $PSBoundParameters
 
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (NikCharlebois) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADNamedLocationPolicy/MSFT_AADNamedLocationPolicy.psm1:398]: Nope it's fixed now.

**Comment by NikCharlebois** [comment]
@ykuijs all comments have been fixed, can you please review again and approve? Thanks

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOBrowserIdleSignout/MSFT_SPOBrowserIdleSignout.psm1 [hunk 4]
```diff
@@ -318,7 +327,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOBrowserIdleSignout/MSFT_SPOBrowserIdleSignout.psm1:331]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOHomeSite/MSFT_SPOHomeSite.psm1 [hunk 5]
```diff
@@ -323,7 +332,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOHomeSite/MSFT_SPOHomeSite.psm1:336]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOHubSite/MSFT_SPOHubSite.psm1 [hunk 6]
```diff
@@ -584,7 +593,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOHubSite/MSFT_SPOHubSite.psm1:597]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOOrgAssetsLibrary/MSFT_SPOOrgAssetsLibrary.psm1 [hunk 7]
```diff
@@ -446,7 +455,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOOrgAssetsLibrary/MSFT_SPOOrgAssetsLibrary.psm1:459]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOPropertyBag/MSFT_SPOPropertyBag.psm1 [hunk 8]
```diff
@@ -365,7 +374,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOPropertyBag/MSFT_SPOPropertyBag.psm1:378]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSearchManagedProperty/MSFT_SPOSearchManagedProperty.psm1 [hunk 9]
```diff
@@ -121,7 +121,10 @@ function Get-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
                 -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSearchManagedProperty/MSFT_SPOSearchManagedProperty.psm1:125]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSearchResultSource/MSFT_SPOSearchResultSource.psm1 [hunk 10]
```diff
@@ -560,7 +569,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSearchResultSource/MSFT_SPOSearchResultSource.psm1:573]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSharingSettings/MSFT_SPOSharingSettings.psm1 [hunk 11]
```diff
@@ -666,7 +675,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSharingSettings/MSFT_SPOSharingSettings.psm1:679]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSite/MSFT_SPOSite.psm1 [hunk 12]
```diff
@@ -897,7 +906,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PnP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSite/MSFT_SPOSite.psm1:910]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteAuditSettings/MSFT_SPOSiteAuditSettings.psm1 [hunk 13]
```diff
@@ -46,7 +46,10 @@ function Get-TargetResource
         -InboundParameters $PSBoundParameters `
         -Url $Url -ErrorAction SilentlyContinue
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteAuditSettings/MSFT_SPOSiteAuditSettings.psm1:50]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteAuditSettings/MSFT_SPOSiteAuditSettings.psm1 [hunk 14]
```diff
@@ -281,7 +290,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteAuditSettings/MSFT_SPOSiteAuditSettings.psm1:294]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteDesign/MSFT_SPOSiteDesign.psm1 [hunk 15]
```diff
@@ -448,7 +457,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteDesign/MSFT_SPOSiteDesign.psm1:461]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1 [hunk 16]
```diff
@@ -376,7 +385,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1:389]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1 [hunk 17]
```diff
@@ -428,7 +437,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOSiteScript/MSFT_SPOSiteScript.psm1:441]: This still exists

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOStorageEntity/MSFT_SPOStorageEntity.psm1 [hunk 18]
```diff
@@ -397,7 +406,10 @@ function Export-TargetResource
         $ConnectionMode = New-M365DSCConnection -Workload 'PNP' `
             -InboundParameters $PSBoundParameters
 
-        #region Telemetry
+        #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
```
[the remark this answers is not in the document]
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOStorageEntity/MSFT_SPOStorageEntity.psm1:410]: This still exists

**Review by ykuijs — CHANGES_REQUESTED** [review]
Not all comments are updated

**Comment by NikCharlebois** [comment]
Weird, I originally did not see these comments. Everything should be aligned now. Thanks

**Review by ykuijs — APPROVED** [review]

**Pull request merged**